### PR TITLE
fix for blank prospective broker agency staff role registration page

### PR DESCRIPTION
**Pull request by TristanB17**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: [187253035](https://www.pivotaltracker.com/n/projects/2640059/stories/187253035)

# A brief description of the changes

Current behavior: prospective broker agency staff role form on the registrations/new page does not render

New behavior: prospective broker agency staff role form on the registrations/new page renders

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [x] DC
- [x] ME

# Additional Context
A view and a feature test removed in [this PR](https://github.com/ideacrew/enroll/pull/3659) were readded

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

**Review by mdkaraman — DISMISSED** [review]
Looks good!  Thank you for adding the clear comments to explain the different use cases for this end point.

**Review by polographer — DISMISSED** [review]

File: components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb [hunk 1]
```diff
@@ -7,15 +7,20 @@ module BrokerAgencies
       class BrokerAgencyStaffRolesController < ::BenefitSponsors::ApplicationController
 
         def new
-          profile = find_broker_agency_profile
-          authorize profile
+          # this endpoint is used for two scenarios
+          # 1.) to render the form for brokers/agents creating new broker agency staff roles for their broker agencies
+          # 2.) to render a different form in the view for benefit_sponsors/profiles/registrations/new for non-users sending in applications to be staff for existing broker agencies
+          # authorization is only enforced here for the first scenario
+          authorize @broker_agency_profile if find_broker_agency_profile
```
Comment (vkghub) [inline components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb:14]: - user is still able to bypass authorization with fake bson id.
- Requested for additional changes

**Review by vkghub — CHANGES_REQUESTED** [review]
Requested for additional changes

File: components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb (hunk 1 above)
Comment (TristanB17) [reply components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb]: changed method to a before action, requested changes made!

**Review by vkghub — APPROVED** [review]
LGTM

**Review by polographer — APPROVED** [review]

**Pull request merged**